lib/board.rb

```diff
@@ -30,4 +29,42 @@ def cells
   def valid_coordinate?(coordinate)
     @cells.key?(coordinate)
   end
+
+  def valid_placement?(ship, coordinates)
+    if !coordinates.empty?
+      coordinates.sort_by! do |coordinate|
+        letter = coordinate[0]
+        number = coordinate[1].to_i
+        [letter, number]
+      end
+
```

Comment:
I moved `!coordinates.empty?` to encapsulate everything inside of the method. I then wrote code to sort the coordinates array being passed in before we do anything with it. This should make it so that the coordinates are set up for our logic every time!

Reply:
Awesome! That makes sense now, so even if the coordinates passed in are not in order, this solves that problem!